browser/importer/brave_profile_writer.cc

```diff
@@ -70,22 +71,31 @@ void BraveProfileWriter::UpdateStats(const BraveStats& stats) {
   }
 }
 
+void BraveProfileWriter::SetBridge(BraveInProcessImporterBridge* bridge) {
+  bridge_ptr_ = bridge;
+}
+
 void BraveProfileWriter::OnRecoverWallet(
     brave_rewards::RewardsService* rewards_service,
     unsigned int result,
     double balance,
     std::vector<brave_rewards::Grant> grants) {
+  rewards_service_->RemoveObserver(this);
+
+  // TODO: check result
   LOG(INFO) << "In RewardsServiceObserver::OnRecoverWallet, result: " << result << ", balance: " << balance;
 
-  // TODO check result
   // TODO: create pref - ready to show pin migrate interface
   //       or "rewards imported" (similar to how Muon has the flag)
+  // ...
 
-  CHECK(!quit_closure_for_wallet_recovery_.is_null());
-  quit_closure_for_wallet_recovery_.Run();
+  // Notify the caller that import is complete
+  bridge_ptr_->FinishLedgerImport();
 }
 
 void BraveProfileWriter::UpdateLedger(const BraveLedger& ledger) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
```

Comment:
this can probably be removed- I accidentally left it in (from when I was experimenting with RunLoop)